### PR TITLE
Update links in readmes

### PR DESCRIPTION
**Pull request by benjaminjb**
**Checklist:**
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**
 - [ ] New feature
 - [ ] Bug fix
 - [x] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

A few links in the readmes no longer worked, as per [this issue](https://github.com/CrunchyData/postgres-operator/issues/3361).

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Other Information**:
Issue [sc-15609]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ Deploy PGO to watch Postgres clusters in all of your [namespaces][k8s-namespaces
 [disaster-recovery]: https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/disaster-recovery/
 [high-availability]: https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/high-availability/
 [monitoring]: https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/monitoring/
-[multiple-cluster]: https://access.crunchydata.com/documentation/postgres-operator/v5/architecture/high-availability/multi-cluster-kubernetes/
+[multiple-cluster]: https://access.crunchydata.com/documentation/postgres-operator/v5/architecture/disaster-recovery/#standby-cluster-overview
```
Comment (benjaminjb) [inline README.md:122]: This page existed in v4, but the closest match in v5 seems to be this other page

File: README.md [hunk 2]
```diff
@@ -174,9 +174,8 @@ Additional containers that are not directly integrated with the PostgreSQL Opera
 
 - [pgPool II](https://access.crunchydata.com/documentation/crunchy-postgres-containers/latest/container-specifications/crunchy-pgpool/)
 - [pg_upgrade](https://access.crunchydata.com/documentation/crunchy-postgres-containers/latest/container-specifications/crunchy-upgrade/)
-- [pgBench](https://access.crunchydata.com/documentation/crunchy-postgres-containers/latest/container-specifications/crunchy-pgbench/)
```
Comment (benjaminjb) [inline README.md:177]: We don't use this container anymore, do we?

File: README.md [hunk 3]
```diff
@@ -240,10 +239,9 @@ When a PostgreSQL Operator general availability (GA) release occurs, the contain
 
 - [Crunchy Data Customer Portal](https://access.crunchydata.com/)
 - [Crunchy Data Developer Portal](https://www.crunchydata.com/developers)
-- [DockerHub](https://hub.docker.com/u/crunchydata)
```
Comment (benjaminjb) [inline README.md:243]: We don't use this anymore.

File: README.md [hunk 4]
```diff
@@ -240,10 +239,9 @@ When a PostgreSQL Operator general availability (GA) release occurs, the contain
...
 
 The image rollout can occur over the course of several days.
 
-To stay up-to-date on when releases are made available in the [Crunchy Data Developer Portal](https://www.crunchydata.com/developers), please sign up for the [Crunchy Data Developer Program Newsletter](https://www.crunchydata.com/developers/newsletter). You can also [join the PGO project community mailing list](https://groups.google.com/a/crunchydata.com/forum/#!forum/postgres-operator/join)
+To stay up-to-date on when releases are made available in the [Crunchy Data Developer Portal](https://www.crunchydata.com/developers), please sign up for the [Crunchy Data Developer Program Newsletter](https://www.crunchydata.com/developers#email). You can also [join the PGO project community mailing list](https://groups.google.com/a/crunchydata.com/forum/#!forum/postgres-operator/join)
```
Comment (benjaminjb) [inline README.md:245]: There's no separate page for newsletter, but there's a tag on the `email` section to sign up for the newsletter at the bottom of the developers page

File: internal/pgbackrest/config.md [hunk 5]
```diff
@@ -25,7 +25,7 @@ During initial cluster creation, four pgBackRest use cases are involved.
 These settings are configured in either the [global] or [stanza] sections of the 
 pgBackRest configuration based on their designation in the pgBackRest code.
 For more information on the above, and other settings, please see
-https://github.com/pgbackrest/pgbackrest/blob/master/src/config/parse.auto.c
+https://github.com/pgbackrest/pgbackrest/blob/release/2.38/src/config/parse.auto.c
```
Comment (benjaminjb) [inline internal/pgbackrest/config.md:28]: As of 5.1.3 and 5.2.0, we moved to pgbackrest 2.40, right? They changed some things and this page was replaced by `parse.auto.c.inc`, but which do we want to point to here?

Comment (benjaminjb) [reply internal/pgbackrest/config.md:28]: We have a few other instances and they all specifically point to the 2.38 page.

File: README.md (hunk 2 above)
Comment (andrewlecuyer) [reply README.md:177]: Honestly, we can remove this "Additional containers" section altogether.  It really doesn't make sense for `pg_upgrade` (which now _is_ integrated w/PGO), and we no longer build `pgBench` or `pgPool` containers for v5.

File: README.md (hunk 1 above)
Comment (andrewlecuyer) [reply README.md:122]: Agree this is the proper v5 equivalent :+1:

File: README.md (hunk 3 above)
Comment (andrewlecuyer) [reply README.md:243]: :+1:

File: internal/pgbackrest/config.md (hunk 5 above)
Comment (andrewlecuyer) [reply internal/pgbackrest/config.md:28]: Yup, agreed.  This should point to the current pgBackRest version we're using for v5 (which should always be the same for all v5 lines).  Good catch!  And something we need to ensure we update when we prep our releases moving forward (assuming the pgBackRest version changes)...

Comment (cbandy) [reply internal/pgbackrest/config.md]: This file contains implementation details, and the links are to the code we based decisions on. I'm not worried about these being up-to-date for users.

File: .golangci.yaml [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 linters:
   disable:
+    - contextcheck
```
Comment (cbandy) [inline .golangci.yaml:5]: ⛏️ Can you please add this to `.golangci.next.yaml` also? One of its gripes seems wrong and one seems right.

**Review by cbandy — APPROVED** [review]
Please squash before merging.

File: README.md (hunk 2 above)
Comment (maaft) [reply README.md:177]: May I ask why you decided to stop build `pgPool` containers?

Is there another solution to how I can split writes and reads? (proxy writes to master and reads to replicas)

**Pull request merged**